public/app/features/panel/panellinks/linkSuppliers.ts

```diff
@@ -1,8 +1,31 @@
 import { PanelModel } from 'app/features/dashboard/state/PanelModel';
-import { FieldDisplay, DataLinkBuiltInVars } from '@grafana/ui';
-import { LinkModelSupplier, getTimeField, ScopedVars } from '@grafana/data';
+import { FieldDisplay } from '@grafana/ui';
+import { LinkModelSupplier, getTimeField, Labels, ScopedVars, ScopedVar } from '@grafana/data';
 import { getLinkSrv } from './link_srv';
 
+interface SeriesVars {
+  name: string;
+  labels: Labels;
+  refId?: string;
+}
+
+interface FieldVars {
+  name: string;
+}
+
+interface ValueVars {
+  raw: number;
+  numeric: number;
+  text: string;
+  time?: number;
+}
+
+interface DataLinkScopedVars extends ScopedVars {
+  __series?: ScopedVar<SeriesVars>;
+  __field?: ScopedVar<FieldVars>;
+  __value?: ScopedVar<ValueVars>;
```

Comment:
can we add `__row?:ScopedVar<any>;`

The row would be backed by DataFrameView and can be accessed either by name or index:
`__row[0]` or `__row.time`

then we could also replace any `__cell_1` variable with `__row[1]`

Reply:
Yes!